### PR TITLE
Fix item-nav shortcut menu ordering

### PR DESCRIPTION
**Pull request by dualcnhq**

### CODE REVIEW

**Comment by iisa** [comment]
thanks for working on this @dualcnhq , let's pu bookmarks before downloads
- TOC
- Search
- Bookmarks
- downloads & the rest

you can repro this by going to [Demo app](https://deploy-preview-1294--lucid-poitras-9a1249.netlify.app/bookreaderdemo/demo-internetarchive?ocaid=theworksofplato01platiala) -> click "Toggle Logged In View" -> add a bookmark -> see bookmark shortcut
and then it should line up properly
image

File: src/BookNavigator/book-navigator.js [hunk 1]
```diff
@@ -71,7 +71,10 @@ export class BookNavigator extends LitElement {
       'volumes',
       'chapters',
       'search',
-      'bookmarks'
+      'downloads',
```
Comment (iisa) [inline src/BookNavigator/book-navigator.js:74]:
```suggestion
      'bookmarks',
      'downloads',
```

File: src/BookNavigator/book-navigator.js [hunk 2]
```diff
@@ -71,7 +71,10 @@ export class BookNavigator extends LitElement {
...
+      'bookmarks',
```
Comment (iisa) [inline src/BookNavigator/book-navigator.js:75]:
```suggestion
```

File: src/BookNavigator/book-navigator.js [hunk 3]
```diff
@@ -379,12 +383,13 @@ export class BookNavigator extends LitElement {
    * Sorts the menuShortcuts property by comparing each provider's id to
    * the id in each iteration over the shortcutOrder array.
    */
-  sortMenuShortcuts() {
-    this.menuShortcuts = this.shortcutOrder.reduce((shortcuts, id) => {
-      const menu = this.menuShortcuts.find((m) => m.id === id);
+  sortMenuShortcuts(menuShortcuts) {
```
Comment (iisa) [inline src/BookNavigator/book-navigator.js:386]: Q: why do we need arg if we are already using instance property?

File: tests/jest/BookNavigator/book-navigator.test.js [hunk 4]
```diff
@@ -325,7 +325,10 @@ describe('<book-navigator>', () => {
             'volumes',
             'chapters',
             'search',
-            'bookmarks'
+            'downloads',
+            'bookmarks',
```
Comment (iisa) [inline tests/jest/BookNavigator/book-navigator.test.js:329]: bookmarks before downloads. 👍

**Review by iisa — CHANGES_REQUESTED** [review]
tweaking order a lil bit:
- volumes
- toc
- search
- bookmarks
- the rest..

+ had a question on updated function

File: src/BookNavigator/book-navigator.js [hunk 5]
```diff
@@ -71,7 +71,10 @@ export class BookNavigator extends LitElement {
       'volumes',
       'chapters',
       'search',
-      'bookmarks'
+      'downloads',
+      'bookmarks',
+      'visualAdjustments',
+      'share'
```
Comment (iisa) [inline src/BookNavigator/book-navigator.js:77]: potential blocker - bc we have these newly listed items here - do they show up as shortcuts?

Comment (iisa) [reply src/BookNavigator/book-navigator.js:77]: we might not need downloads, visualadjustments, share here bc we don't surface them in shortcuts view - let's just remove them.

File: tests/jest/BookNavigator/book-navigator.test.js [hunk 6]
```diff
@@ -325,7 +325,10 @@ describe('<book-navigator>', () => {
             'volumes',
             'chapters',
             'search',
-            'bookmarks'
+            'downloads',
```
Comment (iisa) [inline tests/jest/BookNavigator/book-navigator.test.js:328]:
```suggestion
```

File: tests/jest/BookNavigator/book-navigator.test.js [hunk 7]
```diff
@@ -325,7 +325,10 @@ describe('<book-navigator>', () => {
...
+            'bookmarks',
+            'visualAdjustments',
```
Comment (iisa) [inline tests/jest/BookNavigator/book-navigator.test.js:330]:
```suggestion
```

File: tests/jest/BookNavigator/book-navigator.test.js [hunk 8]
```diff
@@ -325,7 +325,10 @@ describe('<book-navigator>', () => {
...
+            'share'
```
Comment (iisa) [inline tests/jest/BookNavigator/book-navigator.test.js:331]:
```suggestion
```

**Review by iisa — CHANGES_REQUESTED** [review]

**Comment by iisa** [comment]
I have new PR that addresses this, with less change, let's use this - https://github.com/internetarchive/bookreader/pull/1297/files

**Comment by dualcnhq** [comment]
closing this ticket since the fix is already in: https://github.com/internetarchive/bookreader/pull/1297

**Pull request closed without merging**